frontend/src/model/beta/settings/EntityItem.tsx

```diff
@@ -0,0 +1,94 @@
+import ClearIcon from '@mui/icons-material/Clear'
+import GroupsIcon from '@mui/icons-material/Groups'
+import PersonIcon from '@mui/icons-material/Person'
+import { Autocomplete, Chip, IconButton, TableCell, TableRow, TextField, Tooltip, Typography } from '@mui/material'
+import _ from 'lodash-es'
+import { Dispatch, SetStateAction } from 'react'
+import Loading from 'src/common/Loading'
+
+import { useGetModelRoles } from '../../../../actions/model'
+import { CollaboratorEntry, ModelInterface } from '../../../../types/v2/types'
+
+type EntityItemProps = {
+  entity: CollaboratorEntry
+  accessList: CollaboratorEntry[]
+  setAccessList: Dispatch<SetStateAction<CollaboratorEntry[]>>
+  model: ModelInterface
+}
+
+export default function EntityItem({ entity, accessList, setAccessList, model }: EntityItemProps) {
+  const { modelRoles, isModelRolesLoading } = useGetModelRoles(model.id)
+
+  function onRoleChange(_event: React.SyntheticEvent<Element, Event>, newValues: string[]) {
+    const updatedAccessList = _.cloneDeep(accessList)
+    const index = updatedAccessList.findIndex((access) => access.entity === entity.entity)
+    updatedAccessList[index].roles = newValues
+    setAccessList(updatedAccessList)
+  }
+
+  function removeEntity() {
+    setAccessList(accessList.filter((access) => access.entity !== entity.entity))
+  }
+
+  function getModelRoles() {
+    return modelRoles.map((role) => role.id)
+  }
+
+  function getRole(roleId: string) {
+    const role = modelRoles.find((role) => role.id === roleId)
+    if (!role) return { id: roleId, name: 'Unknown Role' }
+
+    return role
+  }
+
+  return (
+    <TableRow>
+      <TableCell>
+        <EntityIcon entity={entity} />
+        <Typography>{entity.entity.replace('user:', '').replace('group:', '')}</Typography>
```

Comment:
Super minor, but could `useMemo` this so it's not being calculated on every render